crates/sc-consensus-subspace/src/verification.rs

```diff
@@ -206,6 +212,20 @@ fn is_within_solution_range(solution: &Solution<FarmerPublicKey>, solution_range
     }
 }
 
+/// Returns true if piece index is within farmer sector
+fn is_within_max_plot(
+    piece_index: PieceIndex,
+    key: &FarmerPublicKey,
+    total_number_of_pieces: u64,
+    max_plot_size: u64,
+) -> bool {
+    if total_number_of_pieces < max_plot_size {
+        return true;
+    }
+    let max_distance = PieceDistance::MAX / total_number_of_pieces * max_plot_size;
```

Comment:
Distance is bidirectional, shouldn't this be twice as small?

Reply:
XOR distance is positive in both directions from node ID on the ring. My understanding is that `PieceDistance::MAX == 0xFFFF...FF`, but for XOR distance the maximum will be just half of that.

Reply:
Actually no, for xor distance maximum will be also `0xFFFF...FF`. In case of piece index hash equal to inverse of identity.

Reply:
Yeah, you're right :nerd_face: